src/Application.vala

```diff
@@ -135,9 +134,29 @@ public class Terminal.Application : Gtk.Application {
         context.set_help_enabled (false);
 
         string[] args = command_line.get_arguments ();
+        string commandline = "";
+        string[] arg_opt = {};
+        string[] arg_cmd = {};
+        bool build_cmdline = false;
+
+        /* Everything after "--" or "-x is to be  treated as a single command to be executed (maybe with its own options)
+         * so it is not passed to the parser. It will be passed as is to a new tab/shell. */
+        foreach (string s in args) {
+            if (s == "--" || s == "-x") {
```

Comment:
You should check here that `build_cmdline` is also false, as it currently is a second -x here would not get parsed as a new tab/shell arg :)

Reply:
Well spotted - thanks!  Now fixed.